### PR TITLE
Fix building OpenEXR with cmake on Windows

### PR DESCRIPTION
**Pull request by meshula**
Incorporates @es0m's fixes, including landing dll's in the bin directory as is conventional on Windows, to make it easy to run openexr programs once they are built. This isn't necessary on platforms with a notion of an rpath, but it is expected on Windows.

see also https://github.com/openexr/openexr/pull/171

### CODE REVIEW

File: IlmBase/Half/CMakeLists.txt [hunk 1]
```diff
@@ -43,9 +43,10 @@ ADD_LIBRARY ( Half ${LIB_TYPE}
 ADD_DEPENDENCIES ( Half toFloat eLut )
 
 INSTALL ( TARGETS
-  Half
-  DESTINATION
-  lib
+  HALF
```
Comment (nickrasmussen) [inline IlmBase/Half/CMakeLists.txt]: Case error here?

File: IlmBase/CMakeLists.txt [hunk 2]
```diff
@@ -158,35 +163,26 @@ IF (NAMESPACE_VERSIONING)
...
-  )
-SET_TARGET_PROPERTIES ( IexMath
-  PROPERTIES
-  VERSION 12.0.0
-  SOVERSION 12
-  OUTPUT_NAME "IexMath${ILMBASE_LIBSUFFIX}"
-  )
+SET (ILMBASE_LIBRARY_NAMES Half Iex Imath IlmThread IexMath)
+FOREACH (LIBNAME ${ILMBASE_LIBRARY_NAMES})
+  IF (LIBNAME STREQUAL "Half")
```
Comment (nickrasmussen) [inline IlmBase/CMakeLists.txt]: Remind me why we’re not suffixing half like all the other libs if the suffix is enabled? Seems like it should be consistent

File: IlmBase/HalfTest/CMakeLists.txt [hunk 3]
```diff
@@ -13,4 +13,9 @@ ADD_EXECUTABLE ( HalfTest
 
 TARGET_LINK_LIBRARIES ( HalfTest Half )
 
+INSTALL ( TARGETS
```
Comment (nickrasmussen) [inline IlmBase/HalfTest/CMakeLists.txt]: Why install targets for the tests?  I thought cmake was able to run them on windows as part of the build process.

File: OpenEXR/IlmImf/CMakeLists.txt [hunk 4]
```diff
@@ -32,8 +40,10 @@ TARGET_LINK_LIBRARIES ( dwaLookups
 
 ADD_CUSTOM_COMMAND (
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dwaLookups.h
+  COMMAND echo "${RUNTIME_DIR}"
```
Comment (nickrasmussen) [inline OpenEXR/IlmImf/CMakeLists.txt]: Remove?

File: OpenEXR/IlmImf/CMakeLists.txt [hunk 5]
```diff
@@ -17,11 +23,13 @@ ADD_CUSTOM_COMMAND (
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/b44ExpLogTable.h
   COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/b44ExpLogTable > ${CMAKE_CURRENT_BINARY_DIR}/b44ExpLogTable.h
   DEPENDS b44ExpLogTable
+  WORKING_DIRECTORY ${RUNTIME_DIR}
 )
 
 ADD_EXECUTABLE ( dwaLookups
   dwaLookups.cpp
 )
+message(STATUS "Using IlmBase version ${ILMBASE_LIBSUFFIX}")
```
Comment (nickrasmussen) [inline OpenEXR/IlmImf/CMakeLists.txt]: If this message is here at all it should use ILMBASE_VERSION as the suffix can be empty

Comment (meshula) [reply OpenEXR/IlmImf/CMakeLists.txt]: debug code :]

**Pull request merged**